internal/backend/telemetry/telemetry.go

```diff
@@ -0,0 +1,140 @@
+package telemetry
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"go.autokitteh.dev/autokitteh/internal/backend/configset"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/exporters/otlp/otlpmetric/otlpmetrichttp"
+	api "go.opentelemetry.io/otel/metric"
+	"go.opentelemetry.io/otel/sdk/metric"
+	"go.opentelemetry.io/otel/sdk/resource"
+	semconv "go.opentelemetry.io/otel/semconv/v1.20.0"
+	"go.uber.org/zap"
+)
+
+type Config struct {
+	Enabled     bool   `koanf:"enabled"`
+	ServiceName string `koanf:"service_name"`
+	Endpoint    string `koanf:"endpoint"`
+}
+
+var Configs = configset.Set[Config]{
+	Default: &Config{Enabled: true, ServiceName: "ak", Endpoint: "localhost:4318"},
+	Dev:     &Config{Enabled: false, ServiceName: "ak", Endpoint: "localhost:4318"},
+}
+
+func verifyConfig(cfg *Config) *Config {
+	if cfg == nil {
+		return Configs.Default
+	}
+	if cfg.ServiceName == "" {
+		cfg.ServiceName = Configs.Default.ServiceName
+	}
+	if cfg.Endpoint == "" {
+		cfg.Endpoint = Configs.Default.Endpoint
+	}
+	return cfg
+}
+
+type Labels map[string]string
+
+func WithLabels(labels Labels) api.MeasurementOption {
```

Comment:
i think if we go and do a `WithLabels` function, it would be more useful to make it just receive a list of strings:

```
func WithLabels(args ...string) api.MeasurementOption {
    var attrs []string
    for len(args) > 1; args = args[2:] {
        attrs = append(attrs, attribute.String(args[0], args[1]))
    }
    return api.WithAttributes(attrs...)
}
```

Much easier to use.

Reply:
accepted your suggestion